src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -97,15 +97,25 @@ public void start(Map<String, String> properties) {
     }
 
     List<String> tables = config.getList(JdbcSourceTaskConfig.TABLES_CONFIG);
+    Boolean tablesFetched = config.getBoolean(JdbcSourceTaskConfig.TABLES_FETCHED);
     String query = config.getString(JdbcSourceTaskConfig.QUERY_CONFIG);
 
     if ((tables.isEmpty() && query.isEmpty())) {
+      // We are still waiting for the tables call to complete.
+      // Start task but do nothing.
+      if (!tablesFetched) {
+        taskThreadId.set(Thread.currentThread().getId());
+        log.info("Started JDBC source task. No work will be done, waiting to fetch tables.");
+        return;
```

Comment:
If we return from here (as opposed to failing) won't this lead to some state not being initialised and throw NPEs when poll is called... So in effect the task might still fail during poll ?